lambdas/get_appointment_blocks.py

```
@@ -0,0 +1,60 @@
+import json
+import boto3
+import datetime
+
+from copy import deepcopy
+from package import db_config
+from package.query_db import query
+from package.lambda_exception import LambdaException
+
+# Author: Victoria Caruso
+
+# function gets all of the appointment blocks for a supporter
+# Input: supporter_id
+# Output: 200 OK
+def lambda_handler(event, context):
+
+    #supporter identifier
+    if 'supporter_id' in event:
+        supporter_id = int(event['supporter_id'])
+    else:
+        raise LambdaException("Invalid input: No user Id")
+
+    #Check if supporter exists
+    sql = "SELECT supporter_id FROM supporters WHERE supporter_id= :supporter_id"
+    supporter_id_param = [{'name' : 'supporter_id', 'value' : {'longValue' : supporter_id}}]
+    existing_supporter = query(sql, supporter_id_param)
+
+    if(existing_supporter['records'] == []):
+        print("No existing Supporter")
+        raise LambdaException("No existing Supporter")
+    
+    #SQL to get all appointments
+    sql = """SELECT start_date, end_date, appointment_block_id, recurring_id FROM appointment_block 
+        WHERE supporter_id = :supporter_id"""
+    sql_parameters = [{'name' : 'supporter_id', 'value': {'longValue' : supporter_id}}]
+    appointment_info = query(sql,sql_parameters)
+        
+    # check if supporter types successfully loaded
+    if (appointment_info['records'] == []):
+        return {
+            'body': json.dumps("There are no appointment blocks for supporter"),
+            'statusCode': 404
```

I believe that you need to raise a lambda exception here.